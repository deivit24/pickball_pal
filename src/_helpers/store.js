import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
