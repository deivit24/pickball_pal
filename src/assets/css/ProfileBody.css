#ProfileBody {
  margin-top: -100px;
  background: white;
  width: 340px;
  padding-bottom: 25px;
  border-radius: 25px;
  box-shadow: 0 0 8px 3px #afafaf;
}

#ProfileBody img {
  border-radius: 50%;
  margin-top: -100px;
  width: 100%;
}

#ProfileBody p {
  font-weight: bold;
}
#ProfileInfo {
  margin-top: 50px;

  max-width: 600px;
}
#ProfileInfo .bio {
  font-weight: 400;
  margin-bottom: 50px;
}
#ProfileInfo h4 {
  font-weight: bold;
  color: #3f50b5;
  margin-top: 25px;
}
#ProfileBody .not-active {
  font-weight: bold;
  color: grey;
}
#ProfileBody .active {
  font-weight: bold;
  color: green;
}
#ProfileBody h3 {
  font-weight: bold;
  color: #3f50b5;
}
