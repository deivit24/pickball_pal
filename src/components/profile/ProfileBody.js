import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import MessageModal from '../messages/MessageModal';
import '../../assets/css/ProfileBody.css';

import { newConversation } from '../../_actions/profile';
const ProfileBody = ({
  avatar,
  first_name,
  last_name,
  bio,
  playing_style,
  active,
  gender,
  level,
}) => {
  const { id } = useParams();
  const auth = useSelector((store) => store.auth);
  const { user } = auth;
  const dispatch = useDispatch();

  function capFirst(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
  }
  const handleSubmit = () => {
    dispatch(newConversation(id));
  };
  let messageMe = false;
  if (!user) {
    messageMe = true;
  }
  let display;
  if (user && user._id === id) {
    display = 'd-none';
  }
  let name = capFirst(first_name) + ' ' + capFirst(last_name);
  return (
    <>
      <div id="ProfileBody" className="container text-center">
        <img src={avatar} className="mx-auto mb-3" alt="" />
        <h3>{name}</h3>
        {active ? (
          <p>
            {' '}
            <i className="fas fa-globe-americas active"></i> Looking to play
          </p>
        ) : (
          <p>
            {' '}
            <i className="fas fa-globe-americas not-active"></i> Not looking to
            play
          </p>
        )}
        <div className="row">
          <div className="col-6">
            <p>{capFirst(gender)}</p>
          </div>
          <div className="col-6">
            <p>Level: {level}</p>
          </div>
          <MessageModal
            messageMe={messageMe}
            display={`${display}`}
            name={name}
            id={id}
            handleSubmit={handleSubmit}
          />
          {/* <Button
            className={`mx-auto ${display}`}
            variant="contained"
            color="primary"
            disabled={messageMe}
            onClick={handleSubmit}
          >
            Message Me
          </Button> */}
        </div>
      </div>
      <div id="ProfileInfo" className="container">
        <div className="row mt-3">
          <div className="col-12 mb-3">
            <h4>About Me</h4>
            <p className="bio mb-3">{bio}</p>
            <h4>Playing Style</h4>
            <p className="bio mb-3">{playing_style}</p>
          </div>
        </div>
      </div>
    </>
  );
};

export default ProfileBody;
