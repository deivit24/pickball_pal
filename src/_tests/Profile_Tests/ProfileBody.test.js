import React from 'react';
import { render } from '@testing-library/react';
import ProfileBody from '../../components/profile/ProfileBody';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing ProfileBody

describe('Testing the ProfileBody', () => {
  let avatar =
      'https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg',
    first_name = 'david',
    last_name = 'salazar',
    bio = 'this is a shiowrt bio',
    playing_style = 'i play hard',
    active = true,
    gender = 'male',
    level = 3.0;

  it('it renders ProfileBody without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfileBody
            avatar={avatar}
            first_name={first_name}
            last_name={last_name}
            bio={bio}
            playing_style={playing_style}
            active={active}
            gender={gender}
            level={level}
          />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfileBody
            avatar={avatar}
            first_name={first_name}
            last_name={last_name}
            bio={bio}
            playing_style={playing_style}
            active={active}
            gender={gender}
            level={level}
          />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
