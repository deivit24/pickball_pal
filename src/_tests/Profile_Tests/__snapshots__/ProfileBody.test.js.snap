// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ProfileBody matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container text-center"
    id="ProfileBody"
  >
    <img
      alt=""
      class="mx-auto mb-3"
      src="https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg"
    />
    <h3>
      David Salazar
    </h3>
    <p>
       
      <i
        class="fas fa-globe-americas active"
      />
       Looking to play
    </p>
    <div
      class="row"
    >
      <div
        class="col-6"
      >
        <p>
          Male
        </p>
      </div>
      <div
        class="col-6"
      >
        <p>
          Level: 3
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained mx-auto undefined MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Message Me
        </span>
      </button>
    </div>
  </div>
  <div
    class="container"
    id="ProfileInfo"
  >
    <div
      class="row mt-3"
    >
      <div
        class="col-12 mb-3"
      >
        <h4>
          About Me
        </h4>
        <p
          class="bio mb-3"
        >
          this is a shiowrt bio
        </p>
        <h4>
          Playing Style
        </h4>
        <p
          class="bio mb-3"
        >
          i play hard
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
